chaosreliably/controls/experiment.py

```diff
@@ -0,0 +1,310 @@
+    """
+    with get_session(configuration, secrets) as session:
+        url = "/entitycontext"
+        resp = session.post(url, content=entity_context.json(by_alias=True))
+        resp.raise_for_status()
+        return entity_context
+
+
+def _create_experiment(
+    experiment_title: str, configuration: Configuration, secrets: Secrets
+) -> EntityContextExperimentLabels:
+    """
+    For a given Experiment title, create a Experiment Entity Context
+    on the Reliably Entity Server.
+
+    :param experiment_title: str representing the name of the Experiment
+    :param configuration: Configuration object provided by Chaos Toolkit
+    :param secrets: Secret object provided by Chaos Toolkit
+    :returns: EntityContextExperimentLabels representing the metadata labels of the
+        created entity - used for `relatedTo` properties in Reliably
+    """
+    experiment_entity = EntityContext(
+        metadata=EntityContextMetadata(
+            labels=EntityContextExperimentLabels(title=experiment_title)
+        )
+    )
+
+    created_entity = _create_entity_context_on_reliably(
+        entity_context=experiment_entity, configuration=configuration, secrets=secrets
+    )
+    return cast(EntityContextExperimentLabels, created_entity.metadata.labels)
+
+
+def _create_experiment_version(
+    commit_hash: str,
+    source: str,
+    experiment_labels: EntityContextExperimentLabels,
+    configuration: Configuration,
+    secrets: Secrets,
+) -> EntityContextExperimentVersionLabels:
+    """
+    For a given commit hash, source link, and Experiment labels, create a
+    ExperimentVersion Entity Context on the Reliably Entity Server.
+
+    :param commit_hash: str representing the SHA1 Hash of the current commit of the
+        Experiments repo at the time of running it
+    :param source: str representing the URL to the source control location
+        of the Experiment being run
+    :param experiment_labels: EntityContextExperimentLabels object representing the
+        labels of the Experiment this version is related to
+    :param configuration: Configuration object provided by Chaos Toolkit
+    :param secrets: Secret object provided by Chaos Toolkit
+    :returns: EntityContextExperimentVersionLabels representing the metadata labels
+        of the created entity - used for `relatedTo` properties in Reliably
+    """
+    experiment_version_entity = EntityContext(
+        metadata=EntityContextMetadata(
+            labels=EntityContextExperimentVersionLabels(
+                commit_hash=commit_hash,
+                source=source,
+            ),
+            related_to=[experiment_labels],
+        )
+    )
+
+    created_entity = _create_entity_context_on_reliably(
+        entity_context=experiment_version_entity,
+        configuration=configuration,
+        secrets=secrets,
+    )
+    return cast(EntityContextExperimentVersionLabels, created_entity.metadata.labels)
```

Comment:
damn, it's really rare to see a call t `cast`

Reply:
Joys of typing..